### ISSUE TITLE
nicer node printing & summary methods

### ISSUE THREAD

**Issue by goldingn** [goldingn/greta#22]
Printing nodes should say that they are greta nodes and preferably state the node type.

For data nodes, which have fixed values, they should summarise the data as if they were arrays (but with `drop = FALSE`). For stochastic or op nodes, they should summarise the data shape and print that with `?` entries.

[This gist](https://gist.github.com/goldingn/343a4c6532f68212110996f44b63e15e) defines an array-like class for unknown values, with subsetting methods etc. This can be used to overload `head()` for nodes. Printing would look like this:

```r
> head(x)
```
```
greta node with unknown values

     [,1] [,2] [,3]
[1,]   ?    ?    ? 
[2,]   ?    ?    ? 
[3,]   ?    ?    ? 
[4,]   ?    ?    ? 
[5,]   ?    ?    ? 
[6,]   ?    ?    ? 
```

A `dim()` method (just returning `x$dim`), would also be helpful.

`summary()` and `str()` methods for distribution nodes should also return some information about the distributions and possibly parameters.

**Comment by goldingn** [comment]
'Node' isn't the best name for these (and it's confusing as nodes are defined internally, too). 'Tensor' may also be confused with TF tensors. Since they are intended to behave like arrays, perhaps 'greta array' is a better term?

The print method might then give:
```
deterministic greta array

     [,1] [,2] [,3]
[1,]  0    1    0 
[2,]  1    0    1.2 
```

```
deterministic greta array with unknown values

     [,1] [,2] [,3]
[1,]   ?    ?    ? 
[2,]   ?    ?    ? 
```

```
stochastic greta array with unknown values

     [,1] [,2] [,3]
[1,]   ?    ?    ? 
[2,]   ?    ?    ? 
```

for consistency, it might be better to thinly wrap these in a `greta_array` S3 class?

**Comment by goldingn** [comment]
ideas for the `greta_array` class are in [this gist](https://gist.github.com/goldingn/5730efe0a4f49760375c71a99b7c933f)

**Comment by goldingn** [comment]
implemented in #24
